bin/package.json

```diff
@@ -1,11 +1,23 @@
 {
   "name": "@glimmer-workspace/bin",
+  "type": "module",
```

Comment:
All packages have `"type": "module"` to reflect reality.

Reply:
this means we can start using publint and arethetypeswrong to ensure we don't have publishing mishaps!

Reply:
yay

Reply:
Yes exactly. This PR also sets up publint (for all published packages) and publint *should* be clean
right now.